src/vdaf/poplar1.rs

```diff
@@ -1245,6 +1246,52 @@ impl<P: Xof<SEED_SIZE>, const SEED_SIZE: usize> Aggregator<SEED_SIZE, 16>
             output_shares,
         )
     }
+
+    /// Validates that no aggregation parameter with the same level as `cur` has been used with the
+    /// same input share before. `prev` contains the aggregation parameters used for the same input.
+    ///
+    /// # Panics
+    /// Panics if `prev.len() > 1`
+    fn is_agg_param_valid(cur: &Poplar1AggregationParam, prev: &[Poplar1AggregationParam]) -> bool {
+        assert!(
+            prev.len() <= 1,
+            "list of previous aggregation parameters must be of size 0 or 1"
+        );
+
+        // Helper function to determine the prefix of `input` at `last_level`.
+        fn get_ancestor(input: &IdpfInput, this_level: u16, last_level: u16) -> IdpfInput {
+            input.prefix((this_level - last_level) as usize)
```

Comment:
This got cleaned up a lot in the spec in cfrg/draft-irtf-cfrg-vdaf#433. The `this_level - last_level` subtraction was only necessary for bit manipulation when we represented all IDPF indices as bigints, so passing that to `prefix()` is incorrect.
```suggestion
        // Helper function to determine the prefix of `input` at `level`.
        fn get_ancestor(input: &IdpfInput, level: u16) -> IdpfInput {
            input.prefix(level as usize + 1)
```